src/library_syscall.js

```diff
@@ -858,7 +858,7 @@ var SyscallsLibrary = {
       var type;
       var name = stream.getdents[idx];
       if (name === '.') {
-        id = stream.id;
+        id = stream.node.id;
```

Comment:
Do all FSes define `stream.node`? Looking in memfs for example I don't think I see that.

That is, I am surprised to see this modify some kind of "ABI" between NODERAWFS and library_fs, on both sides, and not modify all other FSes.

Reply:
As far as I can tell streams are always created via FS.createStream.. except for noderawfs which does its own hacky thing.

From reading the code it looks like streams always contains node, rather than being a node:

https://github.com/emscripten-core/emscripten/blob/272bc3bec4528ce127da4f12152e21c7a04f2177/src/library_fs.js#L1060-L1062

https://github.com/emscripten-core/emscripten/blob/272bc3bec4528ce127da4f12152e21c7a04f2177/src/library_sockfs.js#L78-L80

https://github.com/emscripten-core/emscripten/blob/272bc3bec4528ce127da4f12152e21c7a04f2177/src/library_pipefs.js#L41-L43

You can see the definition of stream assumes that `this.node` is a thing:

https://github.com/emscripten-core/emscripten/blob/272bc3bec4528ce127da4f12152e21c7a04f2177/src/library_fs.js#L405-L413

Also, note that this PR is mostly undoing a previous change that causes this regression: #15167.   My understanding of the FS layer was probably not enough to review that PR.  In particular prior to that change no stream had a `.id` field.   This change restores that previous behaviour, drawing a clear line between a stream and a node.

Reply:
I see, thanks. Yes, looks like streams are created in a special way, which I missed.